extensions/amp-story-shopping/0.1/amp-story-shopping-tag.js

```diff
@@ -96,8 +97,43 @@ export class AmpStoryShoppingTag extends AMP.BaseElement {
     return layout === Layout_Enum.CONTAINER;
   }
 
+  /**
+   * This function counts the number of lines in the shopping tag
+   * and sets the styling properties dynamically based on the number of lines.
+   * @private
+   */
+  styleTagText_() {
+    const pillEl = this.element.shadowRoot?.querySelector(
+      '.amp-story-shopping-tag-pill'
+    );
+
+    const textEl = this.element.shadowRoot?.querySelector(
+      '.amp-story-shopping-tag-pill-text'
+    );
+
+    if (!pillEl || !textEl) {
+      return;
+    }
+
+    const fontSize = parseInt(
+      computedStyle(window, textEl).getPropertyValue('font-size'),
+      10
+    );
+    const ratioOfLineHeightToFontSize = 1.5;
+    const lineHeight = Math.floor(fontSize * ratioOfLineHeightToFontSize);
+    const height = textEl./*OK*/ clientHeight;
+    const numLines = Math.ceil(height / lineHeight);
+
+    if (numLines <= 1) {
+      pillEl.classList.remove('amp-story-shopping-tag-pill-multi-line');
+    } else {
+      pillEl.classList.add('amp-story-shopping-tag-pill-multi-line');
+    }
```

Comment:
Given the adding and removing of classes, this block of code should probably be within a `mutateElement()`

Reply:
Yes, sounds good.